src/db.rs

```diff
@@ -71,6 +71,65 @@ impl DB {
         Ok(())
     }
 
+    pub fn migrate<P: AsRef<Path>>(&mut self, path: P) -> Result<()> {
+        if !self.dirs.is_empty() {
+            bail!(
+                "To prevent conflicts, you can only migrate from z with an empty \
+                 zoxide database!"
+            );
+        }
+
+        let z_db_file = File::open(path).with_context(|| anyhow!("could not open z db file"))?;
+        let reader = BufReader::new(z_db_file);
+
+        for (idx, read_line) in reader.lines().enumerate() {
+            let line_number = idx + 1;
+            let line = if let Ok(line) = read_line {
+                line
+            } else {
+                eprintln!("could not read line {}: {:?}", line_number, read_line);
+                continue;
+            };
+
+            let split_line = line.rsplitn(3, '|').collect::<Vec<&str>>();
+
+            match split_line.as_slice() {
+                [epoch_str, rank_str, path_str] => {
+                    let epoch = epoch_str
+                        .parse::<i64>()
+                        .with_context(|| anyhow!("could not parse epoch: '{}'", epoch_str))?;
```

Comment:
@ajeetdsouza Since we're doing "best effort", should I `eprintln!` these too, instead of erroring? I think I should, but just wanted to make sure.

Reply:
Yes, avoid throwing errors unless they are related to I/O.